src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1660,7 +1660,7 @@ private bool IsPossibleAggregateClauseStartOrStop()
                 || this.IsCurrentTokenWhereOfConstraintClause();
         }
 
-        private BaseListSyntax ParseBaseList()
+        private BaseListSyntax ParseBaseList(SyntaxToken classOrStructOrInterfaceKeyword, bool haveParameters)
```

Comment:
>ParseBaseList [](start = 31, length = 13)

This syntax contradicts the records specification at [https://github.com/dotnet/csharplang/blob/master/proposals/records.md](https://github.com/dotnet/csharplang/blob/master/proposals/records.md). According to the records specification, a class declaration with parameters has no base clause. The grammar appearing in the OP is not used in any other production.

If records are permitted to have base clauses and/or base ctor arguments, please document those (syntax, static semantics, and dynamic semantics) in [https://github.com/dotnet/csharplang/blob/master/proposals/records.md](https://github.com/dotnet/csharplang/blob/master/proposals/records.md) so this PR can be reviewed against the specification. If a record class is permitted to have a base type, please adjust the other sections of the specification to account for that.

Reply:
The syntax is specified in docs\features\records.md and the desired effect of the PR is clearly stated in the description. I think this is good enough to be able to review the PR based on that. Clearly the definition of static semantics, and dynamic semantics is not necessary because this PR isn't trying to implement these aspects of this feature. CC @agocke.

---
In reply to: [432986871](https://github.com/dotnet/roslyn/pull/44733#discussion_r432986871) [](ancestors = 432986871)

Reply:
The document `https://github.com/dotnet/csharplang/blob/master/proposals/covariant-returns.md` is the same one that was moved to the `rejected` folder in `csharplang` recently. Language specifications are kept in `csharplang`.  The current specification for records is at https://github.com/dotnet/csharplang/blob/master/proposals/records.md and there is clearly no base clause in the syntax.

Reply:
> The document [https://github.com/dotnet/csharplang/blob/master/proposals/covariant-returns.md](https://github.com/dotnet/csharplang/blob/master/proposals/covariant-returns.md) is the same one that was moved to the rejected folder in csharplang recently. Language specifications are kept in csharplang.  

I am not sure what point you are trying to make and how covariant-returns.md is related to this PR. The document I referenced above is [https://github.com/dotnet/roslyn/blob/features/records/docs/features/records.md](https://github.com/dotnet/roslyn/blob/features/records/docs/features/records.md)

> The current specification for records is at [https://github.com/dotnet/csharplang/blob/master/proposals/records.md](https://github.com/dotnet/csharplang/blob/master/proposals/records.md) and there is clearly no base clause in the syntax.

I understand that. I believe the PR description is sufficient to draw conclusion about the intent. Could you review this PR assuming that this is the design? Is it not clear? If you have questions on the grammar, I might be able to answer them. If you have arguments against the proposed grammar, this should be taken offline and should not block this PR. We will get necessary approvals and reconcile the specification separately. Otherwise, you always have an option to not review this PR. Thanks. 

---
In reply to: [433022175](https://github.com/dotnet/roslyn/pull/44733#discussion_r433022175) [](ancestors = 433022175)

Reply:
Our process is not to infer the specification from draft implementation, but to check the draft implementation against the specification.

Reply:
> Our process is not to infer the specification from draft implementation, but to check the draft implementation against the specification.

When multiple people sharing work on the feature under tight schedule some things happen in parallel, or out of perfect alignment. 

---
In reply to: [433446995](https://github.com/dotnet/roslyn/pull/44733#discussion_r433446995) [](ancestors = 433446995)